win32ss/user/winsrv/usersrv/harderror.c

```diff
@@ -439,6 +439,58 @@ UserpFormatMessages(
     return Status;
 }
 
+static void
+UserpShowInformationBalloon(
+    PWSTR Text,
+    PWSTR Caption,
+    ULONG ValidResponseOptions,
+    ULONG Status)
+{
+    HWND hShellTrayWnd;
+    COPYDATASTRUCT CopyData;
+    PBALLOON_HARD_ERROR_DATA pdata;
+    DWORD dwSize, cchText, cchCaption;
+    PWCHAR pText, pCaption;
+
+    hShellTrayWnd = FindWindowW(L"Shell_TrayWnd", NULL);
```

Comment:
Nope. This is GetTaskmanWindow(), because other shells that don't have a Shell_TrayWnd but have registered a window using SetTaskmanWindow, should still be able to receive these hard-error notifications.

Reply:
Do you seriously believe there is any third party shell able to use this obscure interface?

Reply:
Yes: test programs on the internet for starters. And perhaps some other 3rd-party shells that I haven't checked their source code.

Reply:
I stand corrected. Windows actually send this to our CTaskSwitchWnd, so GetTaskmanWindow indeed needs to be used.